### ISSUE TITLE
Preview web server listens on public interfaces

### ISSUE THREAD

**Issue by F30** [qvacua/vimr#425]
VimR contains an embedded web server (based on Swifter), which appears to be used (only) for Markdown previews. At the moment, this server listens on all interfaces and is therefore publicly accessible. That is because of the following line in "NetUtils.m":

    sin.sin_addr.s_addr= INADDR_ANY;

This does not publicly expose documents in practice, as the server's port gets chosen at runtime and the exact path depends on the MainWindow's UUID. However, it will definitely be an exploit vector for bugs in Swifter, DoS etc. There might also be paths which don't depend in the UUID, although I didn't find any.

Long story short, I think it would be highly preferable to have the server only listening on localhost.

**Comment by qvacua** [comment]
You're right: That line is not neccessary in `NetUtils`. However, the method `NetUtils.openPort` is only used to determine an unused port. To bind for example only `127.0.0.1` to the socket, I should use some option to do so when `start`ing the `HttpServer` of Swifter. It seems that Swifter hard-coded `INADDR_ANY` when `bind`ing: https://github.com/httpswift/swifter/blob/0a56fec1dbb33c3bf43fe992357c9fbafa0c3b52/Sources/Socket%2BServer.swift#L49

Maybe Swifter 2 already has an option to bind a specific interface? If not, when upgrading to 2, I'll try to do a PR.

**Comment by qvacua** [comment]
I temporarily patched my fork of httpswift/swifter to do this: https://github.com/qvacua/vimr/releases/tag/v0.26.2-305
